src/components/editor/Editor.tsx

```diff
@@ -133,6 +135,14 @@ class Editor extends React.Component<
     if (!this.state.annotationLink) {
       this.setState({ annotationLinkError: 'Musisz podać źródło, jeśli chcesz dodać przypis!' });
       return false;
+    } else if (this.state.annotationLink.length > Editor.linkMaxLength) {
+      this.setState({ annotationLinkError:
+          `Skróć źródło z ${this.state.annotationLink.length} do ${Editor.linkMaxLength} znaków!`,
+      });
+      return false;
+    } else if (!isValidUrl(this.state.annotationLink)) {
```

Comment:
If `isValidUrl` expected protocol presence and if we explicitly normalized it, we could achieve clear separation into "normalization" and "validation" stage (where `isValidUrl` "takes no hostages" -- doesn't assume the https-missing link will be corrected before dispatching it to the server)
We would than validate it as follows:
```js
isValidUrl(httpPrefixed(this.state.annotationLink))
```